src/System.Private.CoreLib/shared/System/Runtime/InteropServices/SuppressGCTransitionAttribute.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// An attribute used to indicate a GC transition should be skipped when making an unmanaged function call.
+    /// </summary>
+    /// <example>
+    /// Example of a valid use case. The Win32 `GetTickCount()` function is a small performance related function
+    /// that reads some global memory and returns the value. In this case, the GC transition overhead is significantly
+    /// more than the memory read.
+    /// <code>
+    /// using System;
+    /// using System.Runtime.InteropServices;
+    /// class Program
+    /// {
+    ///     [DllImport("Kernel32")]
+    ///     [SuppressGCTransition]
+    ///     static extern int GetTickCount();
+    ///     static void Main()
+    ///     {
+    ///         Console.WriteLine($"{GetTickCount()}");
+    ///     }
+    /// }
+    /// </code>
+    /// </example>
+    /// <remarks>
+    /// This attribute is ignored if applied to a method without the <see cref="System.Runtime.InteropServices.DllImportAttribute"/>.
+    ///
+    /// Supressing the GC transition is an advanced scenario and should not be done without fully understanding
+    /// potential consequences. Examples of cases where this attribute is not appropriate:
+    ///   * Native function calls back into the runtime (e.g. Reverse P/Invoke).
+    ///   * Native function performs some type of I/O.
+    ///   * Native function executes for a non-trivial amount of time.
+    ///   * Native function requires extensive object manipulation during marshalling.
```

Comment:
What is example of "extensive object manipulation"?

Reply:
I reworded this in the API review. This should have been phrased as `Native function requires object manipulation that could cause memory allocation during marshalling.`

Reply:
Why is this memory allocation during marshaling a problem? It is not worth it to apply this to methods with complex marshaling because of the performance gain would be miniscule, but it should not be a correctness problem if one applies it to a method with complex marshaling.

Reply:
Okay. I guess I was conflating the two. @stephentoub had a suggestion for better wording for the entire section anyways. Will work on his suggestion.